src/session.rs

```diff
@@ -75,36 +94,40 @@ impl<Connection: io::Read + io::Write + Sync + Send> Session<Connection> {
             Request::ShowPublicKey(ref req) => self.get_public_key(req)?,
             Request::PoisonPill(_req) => return Ok(false),
         };
-        //
+
         let mut buf = vec![];
+
         match response {
             Response::SignedHeartBeat(shb) => shb.encode(&mut buf)?,
             Response::SignedProposal(sp) => sp.encode(&mut buf)?,
             Response::SignedVote(sv) => sv.encode(&mut buf)?,
             Response::Ping(ping) => ping.encode(&mut buf)?,
             Response::PublicKey(pk) => pk.encode(&mut buf)?,
         }
+
         self.connection.write_all(&buf)?;
         Ok(true)
     }
 
     /// Perform a digital signature operation
     fn sign(&mut self, mut request: impl TendermintResponse) -> Result<Response, KmsError> {
         let mut to_sign = vec![];
-        // TODO(ismail): this should either be a config param, or, included in the request!
-        let chain_id = "test_chain_id";
-        request.sign_bytes(chain_id, &mut to_sign)?;
+        request.sign_bytes(self.chain_id.as_str(), &mut to_sign)?;
```

Comment:
Perhaps `sign_bytes` should take a `chain::Id` as an argument? For now I left it a `&str`

Reply:
As long as it is compatible to tendermint (see above), I think using a `chain::Id` would make more sense.